src/instrumenter/instrumentation_context.ts

```diff
@@ -244,4 +247,91 @@ export class InstrumentationContext {
             transCtx.finalize();
         }
     }
+
+    private customMapLibrary = new Map<
+        string,
+        [ContractDefinition, StructDefinition, Map<string, FunctionDefinition>, TypeNode, TypeNode]
+    >();
+
+    isCustomMapLibrary(t: ContractDefinition): boolean {
+        return this.customMapLibrary.has(t.name);
+    }
+
+    getCustomMapLibrary(keyT: TypeNode, valueT: TypeNode): ContractDefinition {
+        const name = getCustomMapLibraryName(keyT, valueT);
+
+        let res = this.customMapLibrary.get(name);
+
+        if (!res) {
+            const library = generateMapLibrary(
+                this.factory,
+                keyT,
+                valueT,
+                this.utilsUnit,
+                this.compilerVersion
+            );
+            const struct = single(library.vStructs);
+            const funcs = new Map(library.vFunctions.map((fn) => [fn.name, fn]));
+            res = [library, struct, funcs, keyT, valueT];
+            this.customMapLibrary.set(name, res);
+        }
+
+        return res[0];
+    }
+
+    getCustomMapStruct(library: ContractDefinition): StructDefinition {
+        const res = this.customMapLibrary.get(library.name);
+        assert(res !== undefined, ``);
+        return res[1];
+    }
+
+    getCustomMapGetter(library: ContractDefinition): FunctionDefinition {
+        const res = this.customMapLibrary.get(library.name);
+        assert(res !== undefined, ``);
```

Comment:
We are startng to get a lot of these empty messages in source. I wonder if it would be better to clarify them or make message argument optional instead. WDYT?

Reply:
Fixed in b4b6be9.